test/integration/iotFleetProvisioning.test.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+const awsRequest = require('@serverless/test/aws-request');
+const { expect } = require('chai');
+const fixtures = require('../fixtures');
+const { deployService, removeService } = require('../utils/integration');
+const { resolveIotEndpoint } = require('../utils/iot');
+
+describe('test/integration/iotFleetProvisioning.test.js', function() {
+  this.timeout(1000 * 60 * 100); // Involves time-taking deploys
+  let stackName;
+  let servicePath;
+  const stage = 'dev';
+
+  const resolveTemplateName = async () => {
+    const result = await awsRequest('CloudFormation', 'describeStacks', { StackName: stackName });
+    return result.Stacks[0].Outputs.find(output => output.OutputKey === 'ProvisioningTemplateName')
+      .OutputValue;
+  };
+
+  before(async () => {
+    let serviceConfig;
+    ({ serviceConfig, servicePath } = await fixtures.setup('iotFleetProvisioning'));
+    const serviceName = serviceConfig.service;
+    stackName = `${serviceName}-${stage}`;
+    await deployService(servicePath);
+  });
+
+  after(async () => {
+    const {
+      certificates: [{ certificateArn, certificateId }],
+    } = await awsRequest('Iot', 'listCertificates');
+    await awsRequest('Iot', 'detachThingPrincipal', {
+      thingName: 'IotDevice',
+      principal: certificateArn,
+    });
+    await awsRequest('Iot', 'detachPolicy', {
+      policyName: 'iotPolicy',
+      target: certificateArn,
+    });
+    await awsRequest('Iot', 'updateCertificate', {
+      certificateId,
+      newStatus: 'INACTIVE',
+    });
+    await awsRequest('Iot', 'deleteCertificate', {
+      certificateId,
+    });
+    await awsRequest('Iot', 'deleteThing', {
+      thingName: 'IotDevice',
+    });
```

Comment:
Can all those operations be done in parallel? Or they need to be done sequentially in given order?